### PR TITLE
Pickling modularization reorg

### PR DESCRIPTION
**Pull request by odersky**
The pickling package got rather large and confusing with three
separate tasks that each had their own conventions: read JVM classfiles,
read Scala2 pickle info, read and write Tasty. The classes for each task are now in
separate packages.

### CODE REVIEW

File: src/dotty/tools/dotc/ast/TreeInfo.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ trait TreeInfo[T >: Untyped <: Type] { self: Trees.Instance[T] =>
   /** Does tree contain an initialization part when seen as a member of a class or trait?
    */
   def isNoInitMember(tree: Tree): Boolean = unsplice(tree) match {
-    case EmptyTree | Import(_, _) | TypeDef(_, _) => true
+    case EmptyTree | Import(_, _) | TypeDef(_, _) | DefDef(_, _, _, _, _) => true
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/ast/TreeInfo.scala]: This seems unrelated to this PR.

**Comment by DarkDimius** [comment]
Otherwise LGTM

**Comment by odersky** [comment]
Yes, that should have been part of some other pull request. I believe the
other one is already in, so this is redundant.

On Sat, May 2, 2015 at 9:56 PM, Dmitry Petrashko <notifications@github.com>
wrote:

> Otherwise LGTM
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lampepfl/dotty/pull/520#issuecomment-98392762>.
>



-- 
Martin Odersky
EPFL
